bundles/org.openhab.binding.teleinfo/src/main/java/org/openhab/binding/teleinfo/internal/handler/TeleinfoAbstractElectricityMeterHandler.java

```diff
@@ -65,19 +65,21 @@ public void initialize() {
 
     @Override
     public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        Bridge bridge = getBridge();
+        TeleinfoAbstractControllerHandler controllerHandler = bridge != null
+                ? (TeleinfoAbstractControllerHandler) bridge.getHandler()
+                : null;
         if (bridgeStatusInfo.getStatus() != ThingStatus.ONLINE) {
+            if (controllerHandler != null) {
+                controllerHandler.removeListener(this);
```

Comment:
You should make sure that this gets called when the TeleinfoAbstractElectricityMeterHandler is disposed as well.
Looks like you need to override the `dispose` method to get that done.

Reply:
Done!